src/encoded/static/components/facets/date_selector.js

```diff
@@ -0,0 +1,353 @@
+                                <select id="select-start-month" value={this.state.startMonth} onChange={this.selectMonth}>
+                                    {this.state.startMonths.map(month =>
+                                        <option value={month} key={month}>{month}</option>
+                                    )}
+                                </select>
+                                <select id="select-start-year" value={this.state.startYear} onChange={this.selectYear}>
+                                    {this.state.startYears.map(year =>
+                                        <option value={year} key={year}>{year}</option>
+                                    )}
+                                </select>
+                            </div>
+                        </div>
+                        <div className="date-arrow">
+                            <i className="icon icon-arrow-right" />
+                        </div>
+                        <div className="date-selector-module">
+                            <h6>End date:</h6>
+                            <div className="date-selector">
+                                <select id="select-end-month" value={this.state.endMonth} onChange={this.selectMonth}>
+                                    {this.state.endMonths.map(month =>
+                                        <option value={month} key={month}>{month}</option>
+                                    )}
+                                </select>
+                                <select id="select-end-year" value={this.state.endYear} onChange={this.selectYear}>
+                                    {this.state.endYears.map(year =>
+                                        <option value={year} key={year}>{year}</option>
+                                    )}
+                                </select>
+                            </div>
+                        </div>
+                    </div>
+                    <div className="date-selector-facet__controls">
+                        <a className="btn btn-info btn-sm apply-date-selector" href={searchString}>Apply changes</a>
+                        <button className="btn btn-info btn-sm reset-date-selector" onClick={() => this.handleReset(resetString)}>
+                            Reset
+                        </button>
+                    </div>
+                </div>
+            );
+        }
+
+        // Facet had all zero terms and was not a "not" facet.
+        return null;
+    }
+}
+
+DateSelectorFacet.propTypes = {
+    /** Relevant `facet` object in `facets` array in `results` */
+    facet: PropTypes.object.isRequired,
+    /** Complete search-results object */
+    results: PropTypes.object.isRequired,
+    /** Query-string portion of current URL without initial ? */
+    queryString: PropTypes.string.isRequired,
+};
+
+DateSelectorFacet.contextTypes = {
+    navigate: PropTypes.func,
+};
+
```

Comment:
I was thinking about the architecture of the facet registry and I feel like maybe most of this date selector code could make more sense in `defaults.js`. Because the date selector facet type is a facet type, and then `date_released` and `date_submitted` are instances of that facet type.

You have a file `datasets_accession.js` which sets parameters for the datasets.accession facet which is an instance of an "exists" facet. And an `nih_institutional_certification.js` file which sets parameters for another instance of an "exists" facet. So to have the same sort of paradigm here, you would have a `date_released.js` and `date_submitted.js` which would set parameters for a "date selector" facet. At least, that is what would make more sense to me. I recognize that there aren't actually any parameters to set for a date selector facet so it's not exactly the same.

Overall, I wonder if this set-up could get a little unwieldy. I really like how all of the default facet types can be all stored in one file and then their parameters can be set elsewhere. But having one file for NIH certification and each specific instance of a default facet type seems like it could get to be a lot of very specific files. Another way to structure it could be to have the `defaults.js` file and then a file for all the instances of "exists" facets and another file for all the instances of the "boolean" facets and "date selector" and "typeahead" facets etc. (Although, here "typeahead" isn't really a default facet type, it's more like an option for the default facet.)

Reply:
I was going to say I don’t mind lots of small files, and I don’t, except with AirBnB who insist on one component per .jsx file, and Redux that goes even finer than that, so I guess I don’t mind — up to a point. But I think you bring up a good point with the datasets_accession and the nih one, and that could (and likely will) happen again. So I’ll rearrange it by type rather than field. Hopefully done after lunch and appointment.